test/forms/buttons/button.test.js

```diff
@@ -1,136 +1,167 @@
 import React, { createRef } from 'react'
-import { mount, shallow } from 'enzyme'
 import { Button } from '../../../src/'
+import { render, screen, act } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 
 test('Button is aria-disabled when form is invalid', () => {
-  const wrapper = shallow(<Button invalid={true}> Hi</Button>)
-  expect(wrapper.props()['aria-disabled']).toBe(true)
+  render(<Button invalid={true}>Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveAttribute('aria-disabled')
 })
 
 test('Button is aria-disabled when form is pristine', () => {
-  const wrapper = shallow(<Button pristine={true}> Hi</Button>)
-  expect(wrapper.props()['aria-disabled']).toBe(true)
+  render(<Button pristine={true}>Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveAttribute('aria-disabled')
 })
 
-test('Button onClick is run when the form is not submitting, pristine, or invalid', () => {
+test('Button onClick is run when the form is not submitting, pristine, or invalid', async () => {
   const onClick = jest.fn()
+
   const formProps = {
     invalid: false,
     pristine: false,
     submitting: false,
   }
-  const wrapper = shallow(
+  render(
     <Button onClick={onClick} {...formProps}>
-      {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const user = userEvent.setup()
```

Comment:
The docs explicitly tell us to run this _before_ the element is rendered.

> We recommend invoking [userEvent.setup()](https://testing-library.com/docs/user-event/setup) before the component is rendered. This can be done in the test itself, or by using a setup function. We discourage rendering or using any userEvent functions outside of the test itself - e.g. in a before/after hook - for reasons described in ["Avoid Nesting When You're Testing"](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing).

https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent

Reply:
Oh I didn't know that 👀